WordPress/src/test/java/org/wordpress/android/ui/reader/discover/interests/ReaderInterestsViewModelTest.kt

```diff
@@ -14,11 +25,54 @@ class ReaderInterestsViewModelTest {
 
     private lateinit var viewModel: ReaderInterestsViewModel
 
+    @Mock lateinit var readerTagRepository: ReaderTagRepository
+
+    @InternalCoroutinesApi
     @Before
     fun setUp() {
-        viewModel = ReaderInterestsViewModel()
+        viewModel = ReaderInterestsViewModel(TEST_DISPATCHER, readerTagRepository)
+    }
+
+    @Test
+    fun `interests ui correctly loaded if non empty interests received from repo on start`() = test {
+        // Given
+        val mockInterests = getMockInterests()
+        whenever(readerTagRepository.getInterests()).thenReturn(mockInterests)
+
+        // When
+        initViewModel()
+
+        // Then
+        assertThat(requireNotNull(viewModel.uiState.value).interests).isEqualTo(mockInterests)
+        assertThat(requireNotNull(viewModel.uiState.value).interestsUiState[0])
+                .isInstanceOf(InterestUiState::class.java)
+        assertThat(requireNotNull(viewModel.uiState.value).interestsUiState[0].title)
+                .isEqualTo(mockInterests[0].tagTitle)
     }
 
     @Test
-    fun `sample test`() {} // TODO:
+    fun `interests ui not updated if no tags received from repo on start`() = test {
```

Comment:
I assume we'll remove this test when we introduce loading/error handling, right?

Reply:
Right.